### PR TITLE
[InAppMessaging] Consolidate FIAM and FIAMSwift

### PR DESCRIPTION
**Pull request by ncooke3**
**Context**
- The contents of FIAM's Swift extension SDK has been moved into the main FIAM module, and the FIAM Swift extension SDK now re-exports the API that used to live in it.
- This change should be non-breaking.

---

**Tasks**
- [x] The affected extension podspecs should be versioned to pin to the releasing version.
- [ ] When this PR is merged, stage the affected podspecs and notify the Games team.

---

**Important**
- This PR contains an assumption in the source code that docs will be at hosted at https://firebase.google.com/docs/ios/swift-migration cc: @morganchen12

### CODE REVIEW

File: FirebaseInAppMessagingSwift.podspec [hunk 1]
```diff
@@ -24,7 +24,7 @@ See more product details at https://firebase.google.com/products/in-app-messagin
   s.prefix_header_file      = false
 
   s.source_files = [
-    'FirebaseInAppMessaging/Swift/Source/**/*.swift',
+    'FirebaseRemoteConfigSwift/Sources/**/*.swift',
   ]
 
   s.test_spec 'unit' do |unit_tests|
```
Comment (paulb777) [inline FirebaseInAppMessagingSwift.podspec:30]: How about changing line 38 to 10.14-beta to minimize the surface area? Also, make a similar change for the other Swift podfiles

File: FirebaseInAppMessagingSwift.podspec [hunk 2]
```diff
@@ -24,7 +24,7 @@ See more product details at https://firebase.google.com/products/in-app-messagin
   s.prefix_header_file      = false
 
   s.source_files = [
-    'FirebaseInAppMessaging/Swift/Source/**/*.swift',
+    'FirebaseRemoteConfigSwift/Sources/**/*.swift',
```
Comment (paulb777) [inline FirebaseInAppMessagingSwift.podspec:27]: RemoteConfig?

Comment (ncooke3) [reply FirebaseInAppMessagingSwift.podspec]: Thanks, fixed.

File: FirebaseInAppMessagingSwift.podspec (hunk 1 above)
Comment (ncooke3) [reply FirebaseInAppMessagingSwift.podspec:30]: good idea!

Comment (ncooke3) [reply FirebaseInAppMessagingSwift.podspec:30]: Shouldn't it be whatever version we release it in? So, 10.15-beta if this goes out for 10.15.0?

Comment (paulb777) [reply FirebaseInAppMessagingSwift.podspec:30]: Yes.

Comment (paulb777) [reply FirebaseInAppMessagingSwift.podspec:30]: Actually 10.16

**Comment by ncooke3** [comment]
Closed and re-opened to deflate some transient network failures in CI.

**Comment by ncooke3** [comment]
I'm a bit unsure of how to fix [`spectesting / specs_testing (FirebaseInAppMessagingSwift.podspec) (pull_request)`](https://github.com/firebase/firebase-ios-sdk/actions/runs/6181098374/job/16778581093?pr=11800).

After reproducing locally, it seems that the `FirebaseInAppMessagingSwift.podspec`'s dependency on `FirebaseInAppMessaging.podspec` isn't coming from the `FirebaseInAppMessagingSwift.podspec` that has been modified in this PR. Therefore, `FirebaseInAppMessagingSwift.podspec` doesn't lint successfully because it really needs the updated `FirebaseInAppMessaging.podspec` that is also changed in this PR.

I'm able to get things linting successfully if I lint the specs together:
```
pod spec lint FirebaseInAppMessagingSwift.podspec FirebaseInAppMessaging.podspec --platforms=ios --skip-tests --allow-warnings --sources=https://github.com/firebase/SpecsTesting,https://cdn.cocoapods.org/ --no-clean
```

I'd imagine this should workflow failure may happen anytime multiple specs are updated in a single PR where the changes are heavily dependent.

Is this considered a bug in the workflow? Should the workflow  lint all changed pod specs together rather than breaking them up into a job matrix to lint each pod spec separately?

**Comment by paulb777** [comment]
It's a known issue we've seen before. Historically, we've ignored interdependent spectesting issues like this until after merging and updating tags.

Let's discuss next week

File: FirebaseInAppMessaging.podspec [hunk 3]
```diff
@@ -38,6 +38,7 @@ See more product details at https://firebase.google.com/products/in-app-messagin
 	base_dir + "Sources/RenderingObjects/**/*.[cmh]",
 	base_dir + "Sources/Runtime/**/*.[cmh]",
 	base_dir + "Sources/Util/**/*.[cmh]",
+  base_dir + "Swift/Source/**/*.swift",
```
Comment (andrewheard) [inline FirebaseInAppMessaging.podspec:41]: nit: Consider making the indentation consistent here (lines 31 - 40 use tabs, line 41 uses 2 spaces, lines 42 - 44 use 4 spaces) while you're touching this file. Alternatively we could look into a ruby formatter in the future and handle them all at once.

**Review by andrewheard — APPROVED** [review]

**Review by paulb777 — APPROVED** [review]
May also want to notify @mikehardy  when the consolidation podspecs are staged.

File: FirebaseInAppMessaging.podspec (hunk 3 above)
Comment (ncooke3) [reply FirebaseInAppMessaging.podspec]: Yep, good call, thanks. Done!

**Pull request merged**